server/src/main/kotlin/core/gameflow/Player.kt

```diff
@@ -1,45 +1,39 @@
 package core.gameflow
 
 import core.Card
+import core.action.bettinground.*
 
+/**
+ * chipsInPot - the amount of player's chips that already went into the pot in previous rbetting rounds,
+ *      not including the chips in player's current bet
+ */
 data class Player(
     val position: Int,
     val stack: Int,
     val holeCards: List<Card> = emptyList(),
+    val bet: Int = 0,
     val chipsInPot: Int = 0,
-    val currentBet: Int = 0,
-    val folded: Boolean = false,
-    val id: Int = position
+    val lastAction: ActionType? = null
 ) {
-    fun withCards(newCards: List<Card>): Player {
-        return this.copy(holeCards = newCards)
+    init {
+        assert(stack >= 0)
+        assert(bet >= 0)
+        assert(chipsInPot >= 0)
     }
 
-    fun afterFold(): Player {
-        return this.copy(folded = true)
-    }
-
-    fun afterAllIn(): Player {
-        return this.copy(currentBet = this.currentBet + this.stack, stack = 0)
-    }
+    val isInGame: Boolean = lastAction != ActionType.FOLD
+    val isAllIn: Boolean = (stack == 0) and isInGame
+    val isDecisive: Boolean = isInGame and !isAllIn
 
-    fun afterRaise(betSize: Int): Player {
-        if (betSize > this.stack)
-            throw NotEnoughChipsException("Bet size of $betSize larger than stack size ${this.stack}")
+    val maxBet: Int = bet + stack
```

Comment:
why do we need this field?

Reply:
Every time we want to check if a player is able to make certain bet, we need to check if they have enough chips in the stack to cover the extra amount that needs to be transfered to their current bet. I think it is better to use this simple helper value than to calculate it manually each time.

Reply:
OK